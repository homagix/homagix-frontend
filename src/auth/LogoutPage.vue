<script setup lang="ts">
import { useRouter } from "vue-router"
import AppButton from "../components/AppButton.vue"

const router = useRouter()
</script>

<template>
  <h2 class="title is-4">Ciao!</h2>
  <p>Wir hoffen, Dich bald wiederzusehen!</p>

  <div class="button-list">
    <AppButton id="recipes-button" @click="router.push('/recipes')">Zu den Rezepten</AppButton>
  </div>
</template>

<style lang="scss">
.button-list {
  margin-top: 1rem;
}
</style>
