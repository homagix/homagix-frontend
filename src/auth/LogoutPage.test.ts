import { describe, expect, it, vi } from "vitest"
import { mount } from "@vue/test-utils"
import LogoutPage from "@/auth/LogoutPage.vue"

const mockRouter = {
  push: vi.fn(),
  back: vi.fn(),
}

vi.mock("vue-router", () => ({ useRouter: () => mockRouter }))

describe("LogoutPage", () => {
  it("should render", () => {
    const wrapper = mount(LogoutPage)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it("should go to recipies page if button is clicked", () => {
    const wrapper = mount(LogoutPage, { global: { mocks: { $router: mockRouter } } })
    wrapper.find("#recipes-button").trigger("click")
    expect(wrapper.vm.$router.push).toHaveBeenCalledWith("/recipes")
  })
})
