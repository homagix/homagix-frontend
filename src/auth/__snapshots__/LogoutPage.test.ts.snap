// Vitest Snapshot v1

exports[`LogoutPage > should render 1`] = `
"<h2 class=\\"title is-4\\">Ciao!</h2>
<p>Wir hoffen, Dich bald wiederzusehen!</p>
<div class=\\"button-list\\"><button id=\\"recipes-button\\" data-v-41d81733=\\"\\">Zu den Rezepten</button></div>"
`;
